v3/src/models/data/data-set.ts

```diff
@@ -464,6 +592,13 @@ export const DataSet = types.model("DataSet", {
         const newCollection = options?.collection ? getGroupedCollection(options.collection) : undefined
         const oldCollection = getCollectionForAttribute(attributeId)
         if (attribute && oldCollection !== newCollection) {
+          if (!attribute.formula.empty) {
+            // If the attribute has a formula, we need to reset all the calculated values to blank values so that they
+            // are not taken into account while calculating case grouping. After the grouping is done, the formula will
+            // be re-evaluated, and the values will be updated to the correct values again.
+            const cases = getCasesForCollection(oldCollection?.id).map(aCase => ({ ...aCase, [attributeId]: "" }))
+            this.setCaseValues(cases)
```

Comment:
Do we need to use `DataSet` API for this? What if there were a `clearValues()` method on the `Attribute` model that accomplished the same thing? I wouldn't think we want/need external clients of the `DataSet` responding to this call to `setCaseValues()` anyway.

Reply:
It's a good point. I had concerns about using some custom methods because I wasn't sure about their safety regarding dataset caching, validations, and other potential helper structures that I might not be aware of yet. I also had doubts about whether the grouping would be calculated correctly. That's why I had used the existing API just to be safe.

However, if you are suggesting this approach, I agree that it sounds like a better option. It involves fewer changes to the DataSet, and it might be faster since there won't be any unnecessary client reactions to the extra 'setCaseValues' call.

I'll go ahead and make that change. Thanks!